packages/mui-base/src/unstable_useNumberInput/useNumberInput.ts

```diff
@@ -440,7 +474,8 @@ export function useNumberInput(parameters: UseNumberInputParameters): UseNumberI
       ...stepperButtonCommonProps,
       disabled: isDecrementDisabled,
       'aria-disabled': isDecrementDisabled,
-      onMouseDown: handleStepperButtonMouseDown,
+      onMouseDown: handleStepperButtonMouseDown('down'),
+      onMouseUp: handleStepperButtonMouseUp,
```

Comment:
Should detect when moving out?


https://github.com/mui/material-ui/assets/3165635/0d598998-0d2c-4fc6-8704-7166098a0ae8